modules/renter/renter.go

```diff
@@ -451,9 +459,24 @@ func NewCustomRenter(g modules.Gateway, cs modules.ConsensusSet, tpool modules.T
 	r.memoryManager = newMemoryManager(defaultMemory, r.tg.StopChan())
 
 	// Load all saved data.
+	id := r.mu.Lock()
 	if err := r.initPersist(); err != nil {
+		r.mu.Unlock(id)
 		return nil, err
 	}
+	r.mu.Unlock(id)
+
+	// Set RenterSettings to persisted data
+	if settingsMetadata.Version != persistVersion {
+		r.persist.MaxDownloadSpeed = DefaultMaxDownloadSpeed
+		r.persist.MaxUploadSpeed = DefaultMaxUploadSpeed
+		r.persist.StreamCacheSize = DefaultStreamCacheSize
+	}
```

Comment:
This block of code should be removed, now it gets handled in the upgrade code.

Reply:
oh wait, maybe I was wrong about that. I can't tell at the moment how the default settings are being established upon the first creation of the renter, but it looks like there's a test demonstrating that they actually do get set correctly.

Reply:
You were right the first time, this code can be removed now that it is in the upgrade code.  `initPersist` is called above which will set the persistence for legacy persistence, and then for current persistence files I just have to read the fields.